src/olympia/blocklist/cron.py

```diff
@@ -39,21 +35,32 @@ def upload_mlbf_to_kinto():
     # https://github.com/mozilla/addons-server/issues/13695
     generation_time = int(time.time() * 1000)
     mlbf = MLBF(generation_time)
+    previous_filter = MLBF(last_generation_time)
+
+    need_mlbf = (
+        last_generation_time < get_blocklist_last_modified_time() or
+        mlbf.blocks_changed_since_previous(previous_filter))
+    if not need_mlbf:
+        log.info(
+            'No new/modified/deleted Blocks in database; '
+            'skipping MLBF generation')
+        return
+
     mlbf.generate_and_write_mlbf()
 
     base_filter_id = get_config(MLBF_BASE_ID_CONFIG_KEY, 0, json_value=True)
-    base_filter = MLBF(base_filter_id) if base_filter_id else None
+    # optimize for when the base_filter was the previous generation so
+    # we don't have to load the blocked JSON file twice.
+    base_filter = (
+        MLBF(base_filter_id)
+        if last_generation_time != base_filter_id else
```

Comment:
Why are we comparing `last_generation_time` to `base_filter_id`? Is that correct? I wouldn't expect to compare a time to an id.

Reply:
`last_generation_time` is used as an id (and `base_filter_id` would be from a time too).  Should I rename `last_generation_time` to make it clearer?